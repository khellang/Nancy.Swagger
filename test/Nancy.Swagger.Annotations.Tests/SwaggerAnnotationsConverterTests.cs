using ApprovalTests;
using ApprovalTests.Reporters;
using Nancy.Swagger.Annotations.Tests.Testdata;
using Nancy.Swagger.Annotations.Tests.Testdata.JsonNetEnricher;
using Nancy.Swagger.Modules;
using Nancy.Swagger.Services;
using Nancy.Testing;
using Xunit;

namespace Nancy.Swagger.Annotations.Tests
{
    [UseReporter(typeof(BeyondCompareReporter))]
    public class SwaggerAnnotationsConverterTests
    {
        private readonly Browser _browser;

        public SwaggerAnnotationsConverterTests()
        {
            var bootstrapper = new ConfigurableBootstrapper(with =>
            {
                with.ApplicationStartup((container, pipelines) =>
                    container.Register<ISwaggerMetadataConverter, SwaggerAnnotationsConverter>());

                with.Module<SwaggerModule>();
                with.Module<TestRoutesModule>();
                with.Module<JsonNetEnricherRoutesModule>();
            });

            _browser = new Browser(bootstrapper);
        }

        [Fact]
        public void Get_ApiDocsPath_ReturnsApiDeclaration()
        {
            ApproveJsonResponse(_browser.Get("/api-docs/api-docs"));
        }

        [Fact]
        public void Get_ApiDocsRootpath_ReturnsResourceListing()
        {
            ApproveJsonResponse(_browser.Get("/api-docs"));
        }

        [Fact]
        public void Get_TestModulePath_ReturnsApiDeclaration()
        {
            ApproveJsonResponse(_browser.Get("/api-docs/testroutes"));
        }

        [Fact]
        public void Get_JsonNetEnricherModulePath_ReturnsApiDeclaration()
        {
            ApproveJsonResponse(_browser.Get("/api-docs/json-net-enricher-routes"));
        }

        private static void ApproveJsonResponse(BrowserResponse response)
        {
            Approvals.VerifyJson(response.Body.AsString());
        }
    }
}