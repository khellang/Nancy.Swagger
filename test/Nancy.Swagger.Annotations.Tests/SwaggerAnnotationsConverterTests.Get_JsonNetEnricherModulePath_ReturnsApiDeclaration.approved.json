{
    "swaggerVersion": "1.2",
    "basePath": "/",
    "apis": [
        {
            "path": "/json-net-enricher-routes/models",
            "operations": [
                {
                    "method": "GET",
                    "nickname": "getJsonNetEnricherRoutesModels",
                    "parameters": [],
                    "type": "array",
                    "items": {
                        "$ref": "JsonNetEnricherModel"
                    }
                }
            ]
        }
    ],
    "models": {
        "JsonNetEnricherModel": {
            "id": "JsonNetEnricherModel",
            "description": "Description of a model",
            "required": [
                "RequiredAlways"
            ],
            "properties": {
                "named": {
                    "type": "string"
                },
                "RequiredAllowNull": {
                    "type": "string"
                },
                "RequiredAlways": {
                    "type": "string"
                },
                "RequiredDefault": {
                    "type": "string"
                }
            }
        }
    }
}